regression/tests/whitelabel/test_dynamic_enrollment_coupon.py

```diff
@@ -0,0 +1,134 @@
+        """
+        super(TestDynamicEnrollmentCoupon, self).setUp()
+        # Initialize common variables
+        course_id, course_info = random.choice(COURSES_CATALOG.items())
+        self.course_id = course_id
+        self.course_price = course_info['price']
+        self.total_price = course_info['price']
+        self.course_title = course_info['title']
+        # Initialize all page objects
+        self.course_about = CourseAboutPage(self.browser, PROF_COURSE_ID)
+
+    def test_enrollment_once_per_customer_code_max_limit(self):
+        """
+        Scenario: Dynamic Enrollment Once Per Customer - Code Max Limit: Each
+        code can be used up to the number of allowed uses and after that it
+        is not usable by any user
+        """
+        self.coupon = Coupon(
+            COURSE_CATALOG_TYPE['multi'],
+            COUPON_TYPE['enroll'],
+            VOUCHER_TYPE['once_per_cust'],
+            catalog_query=CATALOG_QUERY,
+            course_seat_types=COURSE_SEAT_TYPES['prof'],
+            stock_record_ids=[],
+            max_uses=2
+        )
+        self.coupon.setup_coupons_using_api(self.course_price)
+        self.addCleanup(self.coupon.delete_coupon)
+        coupon_code = self.coupon.coupon_codes[0]
+        # Login to application using the existing credentials
+        coupon_users = list(COUPON_USERS.values())
+        last_user = len(coupon_users) - 1
+        for i, coupon_user in enumerate(coupon_users):
+            self.login_and_go_to_basket(coupon_user)
+            if i != last_user:
+                self.addCleanup(
+                    self.unenroll_using_api,
+                    coupon_user,
+                    self.course_id
+                )
+                self.enroll_using_enrollment_code(coupon_code)
+                self.assert_enrollment_and_logout()
+            else:
+                self.assertEqual(
+                    self.error_message_on_invalid_coupon_code(coupon_code),
+                    ONCE_PER_CUSTOMER_CODE_MAX_LIMIT
+                )
+
+    def test_apply_enrollment_once_per_customer_redeem_url(self):
+        """
+        Scenario: Registered Users: Dynamic Enrollment Once Per Customer
+        Redeem URL: Each URL can be used up to the number of allowed uses
+        and after that it is not usable by any user
+        """
+
+        self.coupon = Coupon(
+            COURSE_CATALOG_TYPE['multi'],
+            COUPON_TYPE['enroll'],
+            VOUCHER_TYPE['once_per_cust'],
+            catalog_query=CATALOG_QUERY,
+            course_seat_types=COURSE_SEAT_TYPES['prof'],
+            stock_record_ids=[],
+            max_uses=2
+        )
+        self.coupon.setup_coupons_using_api(self.course_price)
+        coupon_code = self.coupon.coupon_codes[0]
+        # Login to application using the existing credentials
+        coupon_users = list(COUPON_USERS.values())
+        last_user = len(coupon_users) - 1
+        for i, coupon_user in enumerate(coupon_users):
```

Comment:
nit: a way to make this code more readable would be to loop through all but the last user (using list slice notation), and then pull the part that you have out of the else clause and do that separately. Something like:
for coupon_user in coupon_users[:-1]:
    ...
redeem_coupon = RedeemCouponPage...